src/components/side-panel/side-panel.module.ts

```diff
@@ -0,0 +1,21 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+
+import { SidePanelComponent } from './side-panel.component';
+import { SidePanelCloseDirective } from './side-panel-close.directive';
+import { SidePanelService } from './side-panel.service';
+
+const EXPORTS = [
+    SidePanelComponent,
+    SidePanelCloseDirective
+];
+
+@NgModule({
+    imports: [
+        CommonModule
+    ],
+    exports: EXPORTS,
+    declarations: EXPORTS,
+    providers: [SidePanelService]
```

Comment:
Should this service be scoped to the SidePanelComponent instead? If there are multiple side panels I think this would cause them all to open and close whenever any of them did.